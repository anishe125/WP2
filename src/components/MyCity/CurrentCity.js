import React, {Component} from 'react';
import {getHeader, getInfo} from "../../util/City";
import {CurrentCityHeader} from "./CurrentCityHeader";
import {CityInfo} from "../CityInfo";
import Spinner from "react-bootstrap/Spinner";


export class CurrentCity extends Component {

    constructor(props) {
        super(props);
        this.state = {
            cityHeader: '',
            cityInfo: '',
            downloading: true,
            error: false
        };
    }

    render() {
        return (this.state.downloading)
            ? <Spinner animation="border" role="status">
                <span className="sr-only">Loading...</span>
            </Spinner>
            : (this.state.error)
                ? <div className="text-center">
                    <span className="title">Упс, что-то пошло не так</span>
                </div>
                : <div className="row">
                        <CurrentCityHeader data={this.state.cityHeader}/>
                    <div className={"col-6"}>
                        <CityInfo data={this.state.cityInfo}/>
                    </div>
                </div>
    }

    componentDidUpdate(prevProps, prevState, snapshot) {
        if (this.state === prevState) {
            this.getWeather(this.props.position)
        }
    }

    getWeather = async (coords) => {
        this.setState({downloading: true});
        await fetch(`https://api.openweathermap.org/data/2.5/weather?lat=${coords.latitude}&lon=${coords.longitude}&appid=9d5b59f6e7f7e77586dc0269e6d0438a`)
            .then((response) => {
                if (!response.ok) {
                    throw Error(response.statusText);
                }
                return response;
            })
            .then((response) => response.json())
            .then((response) => {
                this.setState({
                    cityHeader: getHeader(response),
                    cityInfo: getInfo(response),
                    error: false,
                    downloading: false
                });
            })
            .catch(err => {
                    this.setState({
                        error: true,
                        downloading: false
                    });
                }
            );

    };


}
