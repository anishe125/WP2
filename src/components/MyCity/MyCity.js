import React  from 'react';
import { connect } from 'react-redux';
import { locRequested } from "../../actions";
import Spinner from "react-bootstrap/Spinner";
import Button from "react-bootstrap/Button";
import Row from "react-bootstrap/Row";
import Col from "react-bootstrap/Col";


const MyCity = ({ city, loading, locRequested, errorMessage }) => {
  const updCityInfo = () => locRequested();

  return (
    <div>
      {loading
        ? <Spinner animation="border" role="status">
            <span className="sr-only">Loading...</span>
          </Spinner>
        : <div>
          <div style={{ marginBottom: 20, marginTop: 10, marginLeft: 70 }}>
            <Button  onClick={updCityInfo}>Обновить геолокацию</Button>
            {errorMessage !== null ? <Row  class="text-lg-center text-danger" style={{  marginTop: 10}}>{errorMessage}</Row> : <Row></Row>}
          </div>
          {city.error
            ? <div>{city.error}</div>
            : <Row>
                <Col sm={6}>
                  <Row>
                    <div class="display-3">{city.city}</div>
                  </Row>

                      <Row >
                        <Col sm={3}><img width={140} src={`http://openweathermap.org/img/wn/${city.icon}@2x.png`} alt=""/></Col>
                        <Col sm={3}><div style={{ marginBottom: 20, marginLeft: 140 , marginRight: 300}} className="display-2">{`${city.temp}°`}</div></Col>
                      </Row>
                    </Col>
                  <Col sm={6}>
                <table class="table">
                  <tr>
                    <td>Pressure:</td>
                    <td align={"right"}>{city.pressure} hPa</td>
                  </tr>
                  <tr>
                    <td>Wind:</td>
                    <td align={"right"}>{city.wind} m/s</td>
                  </tr>
                  <tr>
                    <td>Humidity:</td>
                    <td align={"right"}>{city.humidity}%</td>
                  </tr>
                  <tr>
                    <td>Coordinates:</td>
                    <td align={"right"}>[{city.lat}, {city.lon}]</td>
                  </tr>
                </table>
                </Col>
            </Row>
          }
        </div>
      }
    </div>
  )
};

const mapStateToProps = ({ loading, cityByCoords, errorMessage }) => ({
  city: cityByCoords,
  loading,
  errorMessage
});

export default connect(mapStateToProps, { locRequested })(MyCity);