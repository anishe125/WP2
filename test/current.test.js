import React from 'react';
import {MyCity} from '../src/components/MyCity';
import {shallow} from 'enzyme'
import {shallowToJson} from "enzyme-to-json";

it('renders correctly', () => {
    const tree = shallow(<MyCity/>);
    expect(shallowToJson(tree)).toMatchSnapshot();
});