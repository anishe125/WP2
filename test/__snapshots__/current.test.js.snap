// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <ForwardRef
    noGutters={false}
  >
    <div
      className="display-3 col-5"
    >
      Погода здесь
    </div>
    <div
      className="col-7"
    >
      <Button
        active={false}
        className="btn btn-secondary mt-4 "
        disabled={false}
        onClick={[Function]}
        type="button"
        variant="primary"
      >
        Обновить геопозицию
      </Button>
    </div>
  </ForwardRef>
  <CurrentCity
    position={
      Object {
        "latitude": 55.75,
        "longitude": 37.62,
      }
    }
  />
</div>
`;
